contrib/mac/framework/julia-config.jl

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env julia
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+const options =
+[
+    "--cflags",
+    "--ldflags",
+    "--ldlibs"
+];
```

Comment:
After a quick grep of my changes, the two `DARWIN_FRAMEWORK` variables are only used in base to modify:

1. `LOAD_PATH` points to `[~]/Library/DARWIN_FRAMEWORK_NAME/site/version`.
2. `DL_LOAD_PATH` points to `[~]/Library/DARWIN_FRAMEWORK_NAME/lib`.
3. The shared library name is `DARWIN_FRAMEWORK_NAME` instead of `libjulia[-debug].*`.

The first two points could be made to not depend on the `DARWIN_FRAMEWORK_NAME` and instead always point at `Library/Julia` on macOS, perhaps even when not building as a framework.

The third point could be made OS agnostic by adding a single variable to `build_h.jl` that names `libjulia`.  This would benefit and simplify other places in the code where the library name is constructed by appending `-debug` for debug builds.  Package authors could be recommended to always rely on `julia-config.jl` output if they must build and link against Julia for some reason.